### PR TITLE
fix(theme-classic): code block line number display with line wrapping

### LINKED ISSUE #7907: Inconsistent code block line numbers when word wrap is on
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

When we pass the `showLineNumbers` prop to code block, and we toggle the word wrap button (the code block must have at least one line wider than the code block width), the line numbers are not consistent. Also there are some unintended line breaks when word wrap is on.

### Reproducible demo

_No response_

### Steps to reproduce

1. Initialize new docusaurus project:
```console
npx --yes create-docusaurus@latest website classic
```

2. Let's make one of the code block lines long. Say `docs/tutorial-basics/create-a-page.md`:
```js 
import React from 'react';
import Layout from '@theme/Layout';

export default function MyReactPage() {
  return (
    <Layout>
      <h1>My React page</h1>
      <p>This is a React page. Let's make this sentence bit long. Some more words to make sure...</p>
    </Layout>
  );
}
```
3. Please also include `showLineNumbers` prop to the code block. Serve the website and go to <http://localhost:3000/docs/tutorial-basics/create-a-page>

Word wrap off:
Screenshot 2022-08-06 at 11 01 25 PM

Word wrap on:
Screenshot 2022-08-06 at 11 01 38 PM

### Expected behavior

Line numbers should be consistent. No changes to line numbers. Unnecessary line breaks for lines narrower than the code block width.

### Actual behavior

Inconsistent line numbers after the wrapped line. Also there are line breaks for lines narrower than the code block width.

### Your environment

- Docusaurus version used: 2.0.1
- Environment name and version (e.g. Chrome 89, Node.js 16.4): Safari 15.6, Node.js 16.16.0
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): macOS 12.5

### Self-service

- [ ] I'd be willing to fix this bug myself.

### PR DESCRIPTION
**Pull request by yzhe819**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [x] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [x] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #7907) and the maintainers have approved on my working plan.

## Motivation

Fixes #7907

## Test Plan

Add the dogfooding test page

### Test links
dev link: https://3000-yzhe819-docusaurus-af96kcuq0jz.ws-us59.gitpod.io/tests/docs/code-wrap-test

Deploy preview: https://deploy-preview-7910--docusaurus-2.netlify.app/tests/pages/code-block-tests#code-block-wrapping-tests

## Related issues/PRs

#7907

## Related screenshot
<details>
<summary>screenshot of issue</summary>

图片
</details>

### CODE REVIEW

File: website/_dogfooding/_docs tests/code-wrap-test.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Doc with code
```
Comment (Josh-Cena) [inline website/_dogfooding/_docs tests/code-wrap-test.md:1]: Could you remove this page and add it to here? https://github.com/facebook/docusaurus/blob/main/website/_dogfooding/_pages%20tests/code-block-tests.mdx

Test files are really hard to maintain if they are all over the place

Comment (Josh-Cena) [reply website/_dogfooding/_docs tests/code-wrap-test.md:1]: Actually we already have a code block on this page that demonstrates the issue and I can see that it's fixed:

image

I suggest removing this test. Very much appreciate it that you added it, though.

Comment (yzhe819) [reply website/_dogfooding/_docs tests/code-wrap-test.md:1]: Cool, I will do that.

**Review by Josh-Cena — APPROVED** [review]
Yup. Thanks a lot! Congratulations on your first contribution here!

**Pull request merged**